### ISSUE TITLE
Copy-DbaLogin rename issue

### ISSUE THREAD

**Issue by potatoqualitee** [sqlcollaborative/dbatools#5276]
There are MI issues with Copy-DbaLogin creating the logins and then renaming when using the hash.

**Comment by potatoqualitee** [comment]
look into moving that bit to create

**Comment by Stuart-Moore** [comment]
Can't replicat Jovan's issue, have mailed him asking for a repro

**Comment by potatoqualitee** [comment]
Followed up with Jovan in Slack, will see if it can repro or close.

**Comment by jovanpop-msft** [comment]
Actually this is not a bug it is a feature/refactoring request and probably prerequisite for AAD integration. I will try to explain what is the issue.

I wanted to migrate Windows logins from SQL Server into AAD logins on Managed Instance and I wanted to use `Copy-DbaLogin` and provide `LoginRenameHashtable` parameter that would rename logins, for example EUROPE\jovanpop to AAD jovanpop@microsoft.com login (I also used dynamic switch from Windows to AAD type with the latest SMO that supports AAD login types). The issue that blocked me was the fact that `Copy-DbaLogin` first try to copy logins and then once they are copied it tries to rename the logins. This failed because I cannot create login with windows name in MI. 

I tried to fix this issue by changing the names earlier when the login was created. The idea was to rename logins as soon as possible and then remove part of code that renames logins after migration. Here was the in-progress version:

https://github.com/sqlcollaborative/dbatools/compare/development...JocaPC:development#diff-74fa997065c3079916c19ed42240132d

I wanted first to refactor the code to rename logins as soon as possible, and this would be prerequisite for Win/AAD logins migration.

Unfortunately this didn't worked because after renaming the login I got several errors. It seems that other dependent objects (roles/permission) referenced the login by old name and I could not debug here it happens. I cannot find the repro for this or the error that was returned.

We talked about this issue on MVP summit and this was more like feature/improvement that would enable copying windows to AAD logins. If there is some other way to enable migration of Windows login to AAd logins, ignore my attempt to refactor this.

I guess that the same issue might happen to someone who want sot migrate + rename logins from SQL server to another SQL server where users should change domain names like EUROPE\jovanpop -> USA\jovanpop, but I could not try this.
